dbt/scripts/check_sort_column_names.py

```diff
@@ -0,0 +1,120 @@
+def is_sorted(lst):
+    """Check if a list is sorted, using alphanumeric sorting."""
+    normalized_list = [alphanumeric_key(normalize_string(s)) for s in lst]
+    return normalized_list == sorted(normalized_list)
+
+
+def check_yaml_file(file_path):
+    try:
+        with open(file_path, "r") as file:
+            data = yaml.safe_load(file)
+    except yaml.YAMLError as error:
+        # print(f"Error loading YAML file {file_path}: {e}")
+        return [error], [file_path]  # Return as an error
+
+    def check_columns(data, file_path, unsorted_files_dict, parent_key=None):
+        if isinstance(data, dict):
+            for key, value in data.items():
+                if key == "columns":
+                    column_names = [
+                        col.get("name")
+                        for col in value
+                        if isinstance(col, dict) and "name" in col
+                    ]
+                    if not is_sorted(column_names):
+                        print(f"In file: {file_path}")
+                        print(f"Key above 'columns': {parent_key}")
+                        print("Columns in this group:")
+                        for i, name in enumerate(column_names):
+                            normalized_name = alphanumeric_key(
+                                normalize_string(name)
+                            )
+                            if (
+                                i > 0
+                                and alphanumeric_key(
+                                    normalize_string(column_names[i - 1])
+                                )
+                                > normalized_name
+                            ):
+                                print(f"---> {name}")
+                            else:
+                                print(f"- {name}")
+                        print("-" * 40)  # Separator for clarity
+                        unsorted_files_dict[file_path] += 1  # Increment count
+                else:
+                    check_columns(value, file_path, unsorted_files_dict, key)
+        elif isinstance(data, list):
+            for item in data:
+                check_columns(item, file_path, unsorted_files_dict, parent_key)
+
+    unsorted_files_dict = defaultdict(int)
+    check_columns(data, file_path, unsorted_files_dict)
+    return unsorted_files_dict, []
+
+
+def check_all_yaml_files(directory):
+    unsorted_files_dict = defaultdict(int)
+    error_files = []
+    for root, _, files in os.walk(directory):
+        if "venv" in root:  # Skip virtual environment directories
+            continue
+        for file in files:
+            if file.endswith(".yaml") or file.endswith(".yml"):
+                file_path = os.path.join(root, file)
+                # Temporary to solve the models/model/schema.yaml problem, there are
+                # different partitions of columns under the columns: key
+                # Not sure if it is worth it to build logic to handle that
```

Comment:
The column list in this file is partitioned into different sets of column within the same `columns:` key. This makes the alphanumeric sorting quite complicate. For now my most simple solution is just to remove these files from consideration. I'm curious to hear thoughts on this.